system_tests/speech.py

```diff
@@ -0,0 +1,117 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+
+from google.cloud import exceptions
+from google.cloud import speech
+from google.cloud import storage
+
+from system_test_utils import unique_resource_id
+from retry import RetryErrors
+
+
+class Config(object):
+    """Run-time configuration to be modified at set-up.
+    This is a mutable stand-in to allow test set-up to modify
+    global state.
+    """
+    CLIENT = None
+    TEST_BUCKET = None
+
+
+def setUpModule():
+    Config.CLIENT = speech.Client()
+    # Now create a bucket for GCS stored content.
+    storage_client = storage.Client()
+    bucket_name = 'new' + unique_resource_id()
+    Config.TEST_BUCKET = storage_client.bucket(bucket_name)
+    # 429 Too Many Requests in case API requests rate-limited.
+    retry_429 = RetryErrors(exceptions.TooManyRequests)
+    retry_429(Config.TEST_BUCKET.create)()
+
+
+def tearDownModule():
+    # 409 Conflict if the bucket is full.
+    # 429 Too Many Requests in case API requests rate-limited.
+    bucket_retry = RetryErrors(
+        (exceptions.TooManyRequests, exceptions.Conflict))
+    bucket_retry(Config.TEST_BUCKET.delete)(force=True)
+
+
+class TestSpeechClient(unittest.TestCase):
+    def setUp(self):
+        self.to_delete_by_case = []
+
+    def tearDown(self):
+        for value in self.to_delete_by_case:
+            value.delete()
+
+    def _make_sync_request(self, content=None, source_uri=None,
+                           max_alternatives=None):
+        client = Config.CLIENT
+        sample = client.sample(content=content,
+                               source_uri=source_uri,
+                               encoding=speech.Encoding.LINEAR16,
+                               sample_rate=16000)
+        result = client.sync_recognize(sample,
+                                       language_code='en-US',
+                                       max_alternatives=max_alternatives,
+                                       profanity_filter=True,
+                                       speech_context=['Google',
+                                                       'cloud'])
+        return result
+
+    def _check_best_results(self, results):
+        from google.cloud.speech.transcript import Transcript
+
+        top_result = results[0]
+        self.assertIsInstance(top_result, Transcript)
+        self.assertEqual(top_result.transcript,
+                         'hello thank you for using Google Cloud platform')
+        self.assertGreater(top_result.confidence, 0.90)
+
+    def test_sync_recognize_local_file(self):
+        import os
+        from google.cloud.speech.transcript import Transcript
+
+        file_name = os.path.join('system_tests', 'data', 'hello.wav')
```

Comment:
This is a relative path and will depend on where the file gets run from. You should use `os.path.dirname(__file__)` instead of `system_tests`.

It's almost certainly worth making the value of `os.path.dirname(__file__)` a global and then referring to it when you make paths. But it's also fine to make the actual path to `hello.wav` be a global.